### PR TITLE
tests: drivers: build_all: pwm: Add i2c-devices build test

### PR DESCRIPTION
**Pull request by soburi**
Add build tests for following devices.

- maxim,max31790-pwm
- nxp,pca9685-pwm

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by benediktibk — APPROVED** [review]
Generally a good idea, but it might be easier to maintain in the long run if we switch to something like this: https://github.com/zephyrproject-rtos/zephyr/blob/ac1cc17ae90a9ede2a2b2c5d8311bddf93ea3eca/tests/drivers/build_all/sensor/app.overlay#L74

**Comment by soburi** [comment]
@benediktibk 

> Generally a good idea, but it might be easier to maintain in the long run if we switch to something like this:
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/ac1cc17ae90a9ede2a2b2c5d8311bddf93ea3eca/tests/drivers/build_all/sensor/app.overlay#L74

Thank you for your comment.
There is no consistency in the mechanisms around this, so I will look into it separately.

**Comment by soburi** [comment]
@anangl 
Could you take a look if you have time?

File: tests/drivers/build_all/pwm/boards/native_sim.overlay [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+
+/ {
+	test {
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		test_gpio: gpio@deadbeef {
+			compatible = "vnd,gpio";
+			gpio-controller;
```
Comment (anangl) [inline tests/drivers/build_all/pwm/boards/native_sim.overlay:13]: Do you actually need a GPIO controller here? It does not seem to be used.

**Review by anangl — APPROVED** [review]

**Review by benediktibk — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**